src/Backend/ServiceFactory.php

```diff
@@ -63,6 +66,11 @@ protected function createService($serviceName, array $config)
     {
         if (isset($this->serviceMap[$serviceName])) {
             $service = $this->serviceMap[$serviceName];
+        } elseif (
+            in_array($serviceName, $this->deletedServices)
+            || !file_exists(__DIR__ . '/' . $serviceName . '.php')
+        ) {
+            throw new \InvalidArgumentException('Invalid service name "' . $serviceName . '".');
         } else {
             $serviceClass = 'Heise\\Shariff\\Backend\\'.$serviceName;
```

Comment:
Not sure what is so complicated here. I would simply add a `class_exists()` check below this line and we have all we need.

Reply:
If you mean below the line `$serviceClass = 'Heise\\Shariff\\Backend\\'.$serviceName;` : I'm not sure if this will work because this line was the one which raised a PHP fatal error, and it was not possible to wrap it into a try catch.

Reply:
This is a string concat which for sure did not raise any fatal. ;-)
The line below will fail fatal if the class does not exist. (`new Foo()`)

Reply:
Yes, sorry, mentioned the wrong line. I think I know what you mean. Have to test it here before I implemnent it.

Reply:
But it still needs to make that check cause the calling function to continue the loop. So I decided to raise an invalid argument exception for that purpose, because that is already handled in the desired way in the calling function. Do you think I should do it differently, e.g. returning null in case if class does not exist, and in the calling function add the result to the service map array only if not null? Or is it ok with raising the invalid argument exception?

Reply:
created #137 with my propsal

Reply:
Ok, I will test.

Reply:
Will you fix the codeclimate issue in your PR so I shall wait with testing? OR shall I test it as it is?

Reply:
No, I won't fix this as this is the dumbest rule on earth. (While I agree on the usage of guard clauses, I disagree on this general statement)
Just to give you an idea: There is a rule of thumb for safety critical application "No if without else!". And I take the freedom to state that those safety guys might have more knowledge on good code than those code-climate gals.